src/stk/molecular/molecules/building_block.py

```diff
@@ -644,15 +653,50 @@ def _normalize_placer_ids(self, placer_ids, functional_groups):
         """
 
         if placer_ids is not None:
-            return placer_ids
+            return frozenset(placer_ids)
```

Comment:
If we require the type signature to take a `frozenset` instead of a `tuple` we do not have to do the conversion here.